### PR TITLE
[Trainer] Fix nan-loss condition

### PR DESCRIPTION
**Pull request by anton-l**
Fixes a change in the nan loss handling logic introduced by #13896

With `is_torch_tpu_available()==False` and `(torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step))==False` 
the `else:` condition was never reached, so the training loss was always `0.0`.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Should work fine, thanks for fixing!

**Review by LysandreJik — APPROVED** [review]
Thank you!

**Review by patrickvonplaten — APPROVED** [review]
